firecrest/BasicClient.py

```diff
@@ -654,9 +665,38 @@ def view(self, machine, targetPath):
             "X-Machine-Name": machine,
         }
         params = {"targetPath": targetPath}
-        resp = requests.get(url=url, headers=headers, params=params, verify=self._verify)
+        resp = requests.get(
+            url=url, headers=headers, params=params, verify=self._verify
+        )
         return self._json_response([resp], 200)["output"]
 
+    def whoami(self, sa_role="firecrest-sa"):
+        """Returns the username that FirecREST will be using to perform the other calls.
+        Will return `None` if the token is not valid.
+
+        :param sa_role: this corresponds to the `F7T_AUTH_ROLE` configuration parameter of the site. If you don't know how FirecREST is setup it's better to leave the default.
+        :type sa_role: string
+        :rtype: string or None
+        """
+
+        # FIXME This needs to be added as an endpoint in FirecREST,
```

Comment:
When you refer to "endpoint in FirecREST" do you mean an endpoint like `/utilities/whoami` that actually executes the unix command?

Reply:
I was thinking of `/utilities/whoami` initially that would execute the command and parse the output. But maybe an endpoint in status would be more appropriate. I think what is important is for which user will FirecREST be executing the commands. Since it's just inspecting different claims on the token it's better to be in status maybe.

The are three reasons I don't like this method being in pyFirecREST and this is why I put the FIXME note.
1. There is no way for a user to know what is the configuration of FirecREST to make the call with the right `sa_role` (if we add an endpoint that will be giving configuration parameters it would be better).
2.  We have to duplicate the code of FirecREST, in case at some point it will be accepting a token with different structure this would also need to be updated accordingly.
3. For different versions in the future, FirecREST might be checking something slightly different to find out the username.

Reply:
I can see your points, and I think it's fair to review them.

1. We have an open discussion in our internal backlog to gather more information to show in `/status` endpoint. Maybe a new item could be some information about OIDC (auth endpoint, roles, etc).
2. Yes, in that case, we cannot escape from that.
3. If it's the major version that changes, then we would have to create a new version of pyFirecREST to match the specification of FirecREST.